### ISSUE TITLE
Look into allowing GitHub accounts

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#558]
Look at [GitHub's OAuth page](https://docs.github.com/en/developers/apps/authorizing-oauth-apps) and implement if appropriate.

Currently mb2 doesn't send email, which means people have to contact me to change to have their passwords reset. Supporting email isn't that hard, but considering I already do all my issue tracking via GitHub, it makes sense (to me, at least!) to allow GitHub users to play, especially as I ramp up the number of simultaneous players which I'll do as I work on [ring games](https://github.com/ctm/mb2-doc/issues?q=is%3Aissue+is%3Aopen+label%3A%22ring+games%22) and continuous deployment (#550).

FWIW, the AGDGAT 20210330 component of this issue is just to look at OAuth, so if I look at it and decide to move forward but don't have it working by March 30th, I've still met that goal.

**Comment by ctm** [comment]
I've spent enough time looking into it to decide it's worthwhile to do a spike. I don't know a lot about CORS, but between GitHub's page and a [React tutorial](https://levelup.gitconnected.com/how-to-implement-login-with-github-in-a-react-app-bd3d704c64fc), I can probably either get something working on my own or at least have sufficiently specific questions to get help in an appropriate venue.

If I get all my other agdgat issues done in time to work on the spike this week, I'll leave this ticket open. If not, I'll make a new issue and then update and close this one, because my only goal for this week was to do the research which I've just completed.
